qutebrowser/misc/sessions.py

```diff
@@ -323,7 +331,16 @@ def save_last_window_session(self):
     def _load_tab(self, new_tab, data):
         """Load yaml data into a newly opened tab."""
         entries = []
-        for histentry in data['history']:
+        lazy_load = []
+        # use len(data['history'])
+        # -> dropwhile empty if not session.lazy_session
+        lazy_index = len(data['history'])
+        gen = chain(
+            takewhile(lambda _: not lazy_load, enumerate(data['history'])),
+            enumerate(lazy_load),
+            dropwhile(lambda i: i[0] < lazy_index, enumerate(data['history'])))
```

Comment:
I don't really understand what this does after reading it a few times. Granted, it's 6:30 in the morning 😆 

FWIW I was okay with the previous version, but only if it's guaranteed to actually work. But I suppose this would be okay as well, I just need to read it when I'm more awake :wink: